### PR TITLE
Schema annotations for migrations and objtypes

### PR DESCRIPTION
**Pull request by RobertoPrevato**

### CODE REVIEW

File: edb/schema/links.py [hunk 1]
```diff
@@ -119,6 +119,17 @@ def has_user_defined_properties(self, schema: s_schema.Schema) -> bool:
...
                      if not p.is_special_pointer(schema)])
 
+    def get_source_type(
+        self,
+        schema: s_schema.Schema
+    ) -> s_types.Type:
+        from . import types as s_types
+
+        source = self.get_source(schema)
+        if not isinstance(source, s_types.Type):
```
Comment (elprans) [inline edb/schema/links.py:129]: Let's use an `assert` statement here like we do in other places.

File: edb/schema/objtypes.py [hunk 2]
```diff
@@ -236,12 +257,12 @@ def get_or_create_union_type(
         module=module,
     )
 
-    objtype = schema.get_by_id(type_id, None)
+    objtype: Optional[so.Object] = schema.get_by_id(type_id, None)
```
Comment (elprans) [inline edb/schema/objtypes.py:260]: `get_by_id` also supports the `type` argument, please use that.

File: edb/schema/objtypes.py [hunk 3]
```diff
@@ -257,7 +257,7 @@ def get_or_create_union_type(
         module=module,
     )
 
-    objtype: Optional[so.Object] = schema.get_by_id(type_id, None)
+    objtype = schema.get_by_id(type_id, None, type=so.Object)
```
Comment (elprans) [inline edb/schema/objtypes.py:260]: Why not `ObjectType`?

Comment (RobertoPrevato) [reply edb/schema/objtypes.py]: Sorry, a mistake.

**Pull request merged**